apps/finance/app/src/components/useFilteredTransfers.js

```diff
@@ -0,0 +1,106 @@
+import { useState, useCallback, useMemo } from 'react'
+import { endOfDay, isWithinInterval, startOfDay } from 'date-fns'
+import { TRANSFER_TYPES, Incoming, Outgoing } from '../transfer-types'
+import { addressesEqual } from '../lib/web3-utils'
+
+const UNSELECTED_TOKEN_FILTER = -1
+const UNSELECTED_TRANSFER_TYPE_FILTER = -1
+const UNSELECTED_DATE_RANGE_FILTER = { start: null, end: null }
+
+function useFilteredTransfers({ transactions, tokens }) {
+  const [page, setPage] = useState(0)
+  const [selectedDateRange, setSelectedDateRange] = useState(
+    UNSELECTED_DATE_RANGE_FILTER
+  )
+  const [selectedTransferType, setSelectedTransferType] = useState(
+    UNSELECTED_TRANSFER_TYPE_FILTER
+  )
+  const [selectedToken, setSelectedToken] = useState(UNSELECTED_TOKEN_FILTER)
+  const handleSelectedDateRangeChange = useCallback(range => {
+    setPage(0)
+    setSelectedDateRange(range)
+  }, [])
+  const handleTokenChange = useCallback(index => {
+    setPage(0)
+    setSelectedToken(index || UNSELECTED_TOKEN_FILTER)
+  }, [])
+  const handleTransferTypeChange = useCallback(index => {
+    setPage(0)
+    setSelectedTransferType(index || UNSELECTED_TRANSFER_TYPE_FILTER)
+  }, [])
+  const handleClearFilters = useCallback(() => {
+    setPage(0)
+    setSelectedTransferType(UNSELECTED_TRANSFER_TYPE_FILTER)
+    setSelectedToken(UNSELECTED_TOKEN_FILTER)
+    setSelectedDateRange(UNSELECTED_DATE_RANGE_FILTER)
+  }, [])
+
+  const tokensToFilter = useMemo(() => [{ symbol: 'All tokens' }, ...tokens], [
+    tokens,
+  ])
+
+  const filteredTransfers = useMemo(
+    () =>
+      transactions.filter(({ date, isIncoming, token }) => {
+        const type = isIncoming ? Incoming : Outgoing
+        // Exclude by transaction type
+        if (
+          selectedTransferType !== -1 &&
+          TRANSFER_TYPES[selectedTransferType] !== type
+        ) {
+          return false
+        }
+
+        // Exclude by date range
+        if (
+          // We're not checking for an end date because we will always
+          // have a start date for defining a range.
+          selectedDateRange.start &&
```

Comment:
I mentioned this last time in Agent too, but I think we should make this test more robust to be able to handle any type of input with the date ranges (e.g. has one, has both, has none).

We should treat this hook as being independent with the view, and it will be better practice to handle this state correctly from the get go rather than forgetting to change this (very easy to do!) if we ever change the date range picker's behaviour.

Reply:
Ah yes! This in fact was changed on Agent; this PR was just not updated.